lib/ansible/modules/network/nxos/nxos_file_copy.py

```diff
@@ -60,6 +67,36 @@
       - SSH port to connect to server during transfer of file
     default: 22
     version_added: "2.5"
+  file_pull:
+    description:
+      - Whether or not to pull the remote file from the device. If True,
+        remote file will be copied to local file on the device. If the
+        file already exists on the device, it will be overwritten, and
+        the operation is NOT idempotent.
+    type: bool
+    default: False
+    version_added: "2.7"
+  file_pull_timeout:
+    description:
+      - Use this parameter to set timeout in seconds, when transferring
+        large files or when the network is slow.
+    default: 300
+    version_added: "2.7"
+  remote_scp_server:
+    description:
+      - The remote scp server address which is used to pull the file.
+        This is required if file_pull is True.
+    version_added: "2.7"
+  remote_scp_server_user:
+    description:
+      - The remote scp server username which is used to pull the file.
+        This is required if file_pull is True.
+    version_added: "2.7"
+  remote_scp_server_password:
+    description:
+      - The remote scp server password which is used to pull the file.
+        This is required if file_pull is True.
+    version_added: "2.7"
 '''
 
 EXAMPLES = '''
```

Comment:
Please add specific examples for each of the following.

1. Transfer file from the Ansible server to the NXOS device.
2. Initiate file copy from the NXOS device to copy a file from the SCP server to the NXOS device.

Reply:
Done

Reply:
Done